pkg/apis/eventsource/v1alpha1/types.go

```diff
@@ -417,6 +417,9 @@ type AMQPQueueDeclareConfig struct {
 	// NowWait when true, the queue assumes to be declared on the server
 	// +optional
 	NoWait bool `json:"noWait,omitempty" protobuf:"varint,5,opt,name=noWait"`
+	// Arguments of a queue (also known as "x-arguments") used for optional features and plugins
+	// +optional
+	Arguments []byte `json:"arguments,omitempty" protobuf:"bytes,6,rep,name=arguments"`
```

Comment:
Any reason of using `[]byte`? Are you intended to get it as a base64 encoded string?

Reply:
I am not familiar with protobuf. I know Go well enough to be able to run with things after some guidance, this is a broad question PR which I should be able to fix up with a little guidance.

I spent a fair bit of time trying to figure out what would be most appropriate for the type here. I tried to use Table directly (which I see is a `map[string]interface{}`) which I need to then pass into the `QueueDeclare` method of `Channel` in `amqp/start.go`. This added an import from github.com/streadway/amqp that looked out of place in this types file. Also, the codegen step didn't seem to recognize it.

The `Table` needed in the optional arguments of a queue_declare can be (looking at the pika library in Python) a string key to anything if the typings in Python are correct. That makes sense because these arguments are used for plugins and optional features. It appears I can use a byte slice, which would allow this arbitrary data. Then I can defer the serializing into a Table in the queue declare of the channel.

I'm of course eager to follow any advice on other data types or places to put things.

Reply:
In regards to the message being base64, no, not that I'm aware. I think it is constrained to nearly arbitrary json, except with the top level being a object, or as it's defined in github.com/streadway/amqp, `Table`, which is a `map[string]interface{}`.

[Talking to some people on Slack, starting here](https://gophers.slack.com/archives/C0FSEE1UJ/p1641839086038800) (if you have access to the Gopher Slack), it sounds like getting arbitrary data in protobuf via `[]byte` is probably about the best I can do since the AMQP protocol makes this field maximally dynamic:

> Optional queue arguments, also known as "x-arguments" because of their field name in the AMQP 0-9-1 protocol, is a map (dictionary) of arbitrary key/value pairs that can be provided by clients when a queue is declared.

Then I can serialize it into `Table`, which the AMQP library wants, in the start.go file as I'm doing. There isn't much more I can do with it. A string would cost a slight bit more performance for no real gain.

Reply:
The `arguments` in amqp API is almost an arbitrary map structure, so I'd prefer to put the `arguments` field in the spec as a string in json or yaml format (yaml might be easier):

 ```yaml
      exchangeName: test
      exchangeType: fanout
      routingKey: hello
      queueDeclare:
        name: xxx
        durable: xxxxx
        arguments: |-
          a: b
          c:
            d: e
            f: g
```

Reply:
@whynowy I need a couple more keywords to know what you're asking for, I think...
> The arguments in amqp API is almost an arbitrary map structure
Yep, sounds right to me
> I'd prefer to put the arguments field in the spec as a string in json or yaml format
- spec? Like the Argo types file? Meaning you just want the Arguments type on 422 to be a string here in types.go?
- What are you asking me to do with "in json or yaml format"?
  - Are you asking me to change how the swagger docs are generated? How do I do that?
  - Or are you asking me to change the format sent to Argo from the libraries that I'm using (in my case, Pika in Python)? Or perhaps it's already Yaml... I'm getting the RabbitMQ, argo, and argo-events all set up today to test this out and see if I understand it well enough to make the changes needed.

Reply:
Please use `string` instead of `[]byte` as below.

```
Arguments string `json:"arguments,omitempty" protobuf:"bytes,6,opt,name=arguments"`
```

Reply:
done